### PR TITLE
non_nullable override and some fixes

### PR DESCRIPTION
**Pull request by gkoz**

### CODE REVIEW

File: src/gobjects.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 use std::ascii::AsciiExt;
-use std::collections::HashMap;
+use std::collections::{HashMap};
```
Comment (EPashkin) [inline src/gobjects.rs]: Typo?

**Comment by EPashkin** [comment]
Seems good. Thanks

File: src/codegen/function.rs [hunk 2]
```diff
@@ -81,7 +81,9 @@ pub fn body(env: &Env, analysis: &analysis::functions::Info,
 
     //TODO: change to map on parameters with pass Vec<String> to builder
     for par in &analysis.parameters {
-        let s = par.translate_to_glib(&env.library, in_trait);
+        let upcast = analysis.upcasts.iter().any(|&(ref name, _, _)| name == &par.name)
+            || in_trait && par.instance_parameter;
```
Comment (EPashkin) [inline src/codegen/function.rs]: May be it compiler-optimized, but I prefer check simply condition first.

File: src/gobjects.rs (hunk 1 above)
Comment (gkoz) [inline src/gobjects.rs]: There was `HashSet` there but I replaced it with a sorted vector later. I'll clean this up.

File: src/codegen/function.rs (hunk 2 above)
Comment (gkoz) [inline src/codegen/function.rs]: You're right, thanks for spotting this.

**Comment by gkoz** [comment]
@homu r+

**Pull request merged**